### PR TITLE
feat: expose timeStorageClassUpdated properties of blob's

### PR DESCRIPTION
**Pull request by athakor**

### CODE REVIEW

File: google-cloud-storage/src/main/java/com/google/cloud/storage/BlobInfo.java [hunk 1]
```diff
@@ -882,6 +894,14 @@ public StorageClass getStorageClass() {
     return storageClass;
   }
 
+  /**
+   * Returns the time at which the object's storage class was last changed. When the object is
+   * initially created, it will be set to timeCreated.
+   */
```
Comment (dmitry-fa) [inline google-cloud-storage/src/main/java/com/google/cloud/storage/BlobInfo.java:900]: Returns the time that the object's storage class was last changed or the time of the object creation.

**Review by dmitry-fa — CHANGES_REQUESTED** [review]
You states ` When the object is initially created, it will be set to timeCreated.` But I don't see the code which set this value to `timeCreated`.

**Comment by athakor** [comment]
> You states ` When the object is initially created, it will be set to timeCreated.` But I don't see the code which set this value to `timeCreated`.

this thing done through the APi. you can check the confirmance test's here https://github.com/athakor/java-storage/pull/19/files#diff-8c18a40b13cc5c5d90547c273470f5eeR3572

**Comment by dmitry-fa** [comment]
> 
> 
> > You states ` When the object is initially created, it will be set to timeCreated.` But I don't see the code which set this value to `timeCreated`.
> 
> this thing done through the APi. you can check the confirmance test's here https://github.com/athakor/java-storage/pull/19/files#diff-8c18a40b13cc5c5d90547c273470f5eeR3572

okay, that makes sense.
But I have another question:
```
    // You can't change an object's storage class directly, the only way is to rewrite the object
    // with the desired storage class.
```
When `timeStorageClassLastUpdated` differs from `timeCreated`?

**Comment by athakor** [comment]
> > > You states ` When the object is initially created, it will be set to timeCreated.` But I don't see the code which set this value to `timeCreated`.
> > 
> > 
> > this thing done through the APi. you can check the confirmance test's here https://github.com/athakor/java-storage/pull/19/files#diff-8c18a40b13cc5c5d90547c273470f5eeR3572
> 
> okay, that makes sense.
> But I have another question:
> 
> ```
>     // You can't change an object's storage class directly, the only way is to rewrite the object
>     // with the desired storage class.
> ```
> 
> When `timeStorageClassLastUpdated` differs from `timeCreated`?

when you change the object's storage class you can verify here https://github.com/athakor/java-storage/pull/19/files#diff-8c18a40b13cc5c5d90547c273470f5eeR3576-R3587

**Comment by dmitry-fa** [comment]
you compare two blobs,
but I'm afraid for the for the same blob  timeStorageClassUpdated == timeCreated

**Comment by athakor** [comment]
> you compare two blobs,
> but I'm afraid for the for the same blob timeStorageClassUpdated == timeCreated

these is based on Api and proto model so i dont think so we need to worry about it.

File: google-cloud-storage/src/main/java/com/google/cloud/storage/BlobInfo.java (hunk 1 above)
Comment (athakor) [reply google-cloud-storage/src/main/java/com/google/cloud/storage/BlobInfo.java:900]: done

**Review by pmakani — APPROVED** [review]

**Comment by dmitry-fa** [comment]
we can just introduce a getter returning the timeCreated value with the same effect. I don't see a reason to spoil the code.

**Comment by dmitry-fa** [comment]
Maybe you can ask Frank about that.

**Comment by athakor** [comment]
> we can just introduce a getter returning the timeCreated value with the same effect. I don't see a reason to spoil the code.

What do you mean..to spoil the code..If there is no reason then why they introduced in api proto model..??

**Pull request closed without merging**